evm_arithmetization/src/cpu/kernel/asm/core/access_lists.asm

```diff
@@ -63,17 +63,22 @@ global init_access_lists:
     POP
 %endmacro
 
-// Multiply the ptr at the top of the stack by 2
-// and abort if 2*ptr - @SEGMENT_ACCESSED_ADDRESSES >= @GLOBAL_METADATA_ACCESSED_ADDRESSES_LEN
-// In this way ptr must be pointing to the begining of a node.
+// Multiply the value at the top of the stack, denoted by ptr/2, by 2
+// and abort if ptr/2 >= mem[@GLOBAL_METADATA_ACCESSED_ADDRESSES_LEN]/2
+// In this way 2*value must be pointing to the begining of a node.
 %macro get_valid_addr_ptr
-    // stack: ptr
+    // stack: ptr/2
+    DUP1
+    // stack: ptr/2, ptr/2
+    %mload_global_metadata(@GLOBAL_METADATA_ACCESSED_ADDRESSES_LEN)
+    // @GLOBAL_METADATA_ACCESSED_ADDRESSES_LEN must be an even number because
+    // both @SEGMENT_ACCESSED_ADDRESSES and the unscaled access addresses list len
+    // must be even numbers
+    %div_const(2)
+    // stack: SEGMENT_ACCESSED_ADDRESSES/2 + access_addr_list_len/2, ptr/2, ptr/2
```

Comment:
nit: I get that you're trying to highlight the length is used as offset in the memory address, but I find the `SEGMENT_ACCESSED_ADDRESSES/2 + access_addr_list_len/2` a bit confusing. Couldn't we just write `scaled_len`, eventually explaining in short comment that it's `segment + list_len`?

Reply:
`scaled_len/2`?

Reply:
Yeah I meant the name before the `%div_const(2)` application. Would be happy with `scaled_len/2`!